### ISSUE TITLE
setup fails at ghcjs-boot --dev under linux (first missing primpops-txt.js, then ghc 7.10.1 single-file package database issue)

### ISSUE THREAD

**Issue by asciiascetic** [ghcjs/ghcjs#338]
$ ghc --numeric-version
    7.10.1
    $ cabal --numeric-version
    1.22.2.0

The `cabal install ./ghcjs ./ghcjs-prim` step seems to have finished without complaint.  And then

    $ ghcjs-boot --dev
    .... (much omitted, and then) ...
    Resolving dependencies...
    [1 of 1] Compiling Main             ( boot/ghc-prim/dist/setup/setup.hs, boot/ghc-prim/dist/setup/Main.o )
    
    boot/ghc-prim/dist/setup/setup.hs:19:1: Warning:
        Module ‘System.Cmd’ is deprecated: Use "System.Process" instead
    Linking ./boot/ghc-prim/dist/setup/setup ...
    Configuring ghc-prim-0.4.0.0...
    /bin/sh: 1: cannot open ../../data/primops-js.txt: No such file
    Failed to install ghc-prim-0.4.0.0
    cabal: Error: some packages failed to install:
    ghc-prim-0.4.0.0 failed during the building phase. The exception was:
    ExitFailure 2
    ghcjs-boot: error running: /opt/cabal/1.22/bin/cabal --config-file /home/third/.ghcjs/x86_64-linux-0.1.0-7.10.1/ghcjs/cabalBootConfig install ./boot/ghc-prim --solver=topdown --global --ghcjs --one-shot --avoid-reinstalls --builddir dist --with-compiler /home/third/.cabal/bin/ghcjs --with-hc-pkg /home/third/.cabal/bin/ghcjs-pkg --prefix /home/third/.ghcjs/x86_64-linux-0.1.0-7.10.1/ghcjs --enable-documentation --haddock-html --haddock-hyperlink-source --enable-library-profiling --root-cmd /home/third/.ghcjs/x86_64-linux-0.1.0-7.10.1/ghcjs/run.sh
        exit status: 1
        stderr: Warning: No remote package servers have been specified. Usually you would have
        one specified in the config file.
        
        boot/ghc-prim/dist/setup/setup.hs:19:1: Warning:
            Module ‘System.Cmd’ is deprecated: Use "System.Process" instead
        /bin/sh: 1: cannot open ../../data/primops-js.txt: No such file
        cabal: Error: some packages failed to install:
        ghc-prim-0.4.0.0 failed during the building phase. The exception was:
        ExitFailure 2

**Comment by manny-fp** [comment]
I hope you don't mind that I'm piggybacking on your issue, but I'm having a similar problem (albeit with GHC 7.8.4 instead of 7.10.1), in case they're related.

```
$ ghc --version
The Glorious Glasgow Haskell Compilation System, version 7.8.4
$ cabal --version
cabal-install version 1.22.3.0
using version 1.22.3.0 of the Cabal library
$ ghcjs-boot --dev
[...]
Configuring transformers-0.4.3.0...
Building transformers-0.4.3.0...
Preprocessing library transformers-0.4.3.0...
[ 1 of 28] Compiling Data.Functor.Identity ( oldsrc/Data/Functor/Identity.hs, dist/build/Data/Functor/Identity.js_o )
[ 2 of 28] Compiling Control.Monad.Trans.Class ( Control/Monad/Trans/Class.hs, dist/build/Control/Monad/Trans/Class.js_o )
[ 3 of 28] Compiling Control.Monad.Signatures ( Control/Monad/Signatures.hs, dist/build/Control/Monad/Signatures.js_o )
[ 4 of 28] Compiling Control.Monad.IO.Class ( Control/Monad/IO/Class.hs, dist/build/Control/Monad/IO/Class.js_o )
[ 5 of 28] Compiling Control.Monad.Trans.Cont ( Control/Monad/Trans/Cont.hs, dist/build/Control/Monad/Trans/Cont.js_o )
[ 6 of 28] Compiling Control.Monad.Trans.Reader ( Control/Monad/Trans/Reader.hs, dist/build/Control/Monad/Trans/Reader.js_o )
[ 7 of 28] Compiling Control.Monad.Trans.RWS.Lazy ( Control/Monad/Trans/RWS/Lazy.hs, dist/build/Control/Monad/Trans/RWS/Lazy.js_o )
[ 8 of 28] Compiling Control.Monad.Trans.RWS ( Control/Monad/Trans/RWS.hs, dist/build/Control/Monad/Trans/RWS.js_o )
[ 9 of 28] Compiling Control.Monad.Trans.RWS.Strict ( Control/Monad/Trans/RWS/Strict.hs, dist/build/Control/Monad/Trans/RWS/Strict.js_o )
[10 of 28] Compiling Control.Monad.Trans.State.Lazy ( Control/Monad/Trans/State/Lazy.hs, dist/build/Control/Monad/Trans/State/Lazy.js_o )
[11 of 28] Compiling Control.Monad.Trans.State ( Control/Monad/Trans/State.hs, dist/build/Control/Monad/Trans/State.js_o )
[12 of 28] Compiling Control.Monad.Trans.State.Strict ( Control/Monad/Trans/State/Strict.hs, dist/build/Control/Monad/Trans/State/Strict.js_o )
[13 of 28] Compiling Data.Functor.Classes ( Data/Functor/Classes.hs, dist/build/Data/Functor/Classes.js_o )
Data/Functor/Classes.hs:180:10: Not in scope: ‘mappend’
Data/Functor/Classes.hs:197:47: Not in scope: ‘mappend’
Data/Functor/Classes.hs:221:21: Not in scope: ‘mappend’
Data/Functor/Classes.hs:261:41: Not in scope: ‘mappend’
Failed to install transformers-0.4.3.0
[...]
transformers-0.4.3.0 failed during the building phase. The exception was:
ExitFailure 1
```

**Comment by asciiascetic** [comment]
Not at all! Though I wonder if they aren't separate issues. 

The trouble I'm having has to do with a code preprocessing step either failing silently or writing its output to the wrong location.  Either way, it cannot be found by the build process.

**Comment by asciiascetic** [comment]
@manny-fp Also, check out the README

> Try --constraint=transformers==0.3.0.0 (or the version that came with your GHC) if the problem looks related to this package.

Since you're having trouble building the transformer's package, maybe its a simple matter of figuring out the right version and adding in the correct constraint.  I see that in your build, the version number is already ahead of what the README suggests, maybe it's too new?

**Comment by BethAr** [comment]
My new issue seems quite related to the OP, albeit on windows:
```
Resolving dependencies...
[1 of 1] Compiling Main             ( boot\ghc-prim\dist\setup\setup.hs, boot\ghc-prim\dist\setup\Main.o )

boot\ghc-prim\dist\setup\setup.hs:19:1: Warning:
    Module `System.Cmd' is deprecated: Use "System.Process" instead
Linking ./boot/ghc-prim\dist\setup\setup.exe ...
Configuring ghc-prim-0.4.0.0...
The system cannot find the file specified.
Failed to install ghc-prim-0.4.0.0
cabal.exe: Error: some packages failed to install:
ghc-prim-0.4.0.0 failed during the building phase. The exception was:
ExitFailure 1
ghcjs-boot.exe: error running: C:\Users\Beta\AppData\Roaming\cabal\bin\cabal.exe --config-file C:\Users\Beta\AppData\Roaming\ghcjs\i386-mingw32-0.1.0-7.10.1\ghcjs\cabalBootConfig install ./boot/ghc-prim --solver=topdown --configure-option=--with-gmp-includes=C:\Users\Beta\AppData\Roaming\ghcjs\i386-mingw32-0.1.0-7.10.1\ghcjs\ghcjs-boot\boot\integer-gmp\gmp\intree\include --configure-option=--with-intree-gmp --global --ghcjs --one-shot --avoid-reinstalls --builddir dist --with-compiler C:\Users\Beta\AppData\Roaming\cabal\bin\ghcjs.exe --with-hc-pkg C:\Users\Beta\AppData\Roaming\cabal\bin\ghcjs-pkg.exe --prefix C:\Users\Beta\AppData\Roaming\ghcjs\i386-mingw32-0.1.0-7.10.1\ghcjs --enable-documentation --haddock-html --haddock-hyperlink-source --enable-library-profiling --disable-executable-stripping --disable-library-stripping
exit status: 1
stderr: Warning: No remote package servers have been specified. Usually you would have
one specified in the config file.

boot\ghc-prim\dist\setup\setup.hs:19:1: Warning:
    Module `System.Cmd' is deprecated: Use "System.Process" instead
The system cannot find the file specified.
cabal.exe: Error: some packages failed to install:
ghc-prim-0.4.0.0 failed during the building phase. The exception was:
ExitFailure 1

```

**Comment by asciiascetic** [comment]
UPDATE:

I had initially run `ghcjs-boot --dev`, encountered error, and tried again with `ghcjs-boot --dev --ghcjs-boot-dev-branch ghc-7.10` as suggested in the README.  Both produced the same error reported above.

So I decided to `rm -rf ~/.ghcjs` and rebuild before trying that second command.  I no longer get the error regarding the missing `primops-js.txt` file, but the process still fails.  

I might consider this closed and open up a new issue.

**Comment by BethAr** [comment]
When you say _rebuild_, do you mean GHCJS or only the GHCJS Libraries?

**Comment by asciiascetic** [comment]
@BethAr The former. 

    cd ~/
    rm -rf .ghcjs
    cabal install ./ghcjs ./ghcjs-prim
    ghcjs-boot --dev ----ghcjs-boot-dev-branch ghc-7.10

That did not produce the error regarding the missing file, and so in that sense, the issue is resolved.

HOWEVER, the boot did not finish, and I got a different error about a unsupported single-file package databases.  I may open up a new issue.

**Comment by BethAr** [comment]
From GHC 7.10.1 release notes: 
>1.5.2.7. Package system

>ghc-pkg (and ghc) have dropped support for single-file style package databases. Since version 6.12, >ghc-pkg has defaulted to a new database format (using a directory of files, one per package plus a >binary cache).

Maybe it is related?

**Comment by asciiascetic** [comment]
Yeah so [here](https://github.com/ghcjs/ghcjs-boot/issues/15 ) the unsupported package database issue is discussed.

**Comment by BethAr** [comment]
I also got the database error this time, after deleting the /ghcjs folder. I will add voice to that issue.
```
boot\ghc-prim\dist\setup\setup.hs:19:1: Warning:
    Module `System.Cmd' is deprecated: Use "System.Process" instead
ghcjs.exe: ghc no longer supports single-file style package databases (dist\package.conf.inplace) use 'ghc-pkg init' to create the database with the correct format.
cabal.exe: Error: some packages failed to install:
ghc-prim-0.4.0.0 failed during the building phase. The exception was:
ExitFailure 1
```

**Comment by asciiascetic** [comment]
I for one am content to wait until ghc 7.10.2 is released.

**Comment by manny-fp** [comment]
[My problem](https://github.com/ghcjs/ghcjs/issues/338#issuecomment-98699975) was totally separate, it turns out that the `transformers` commit referenced by https://github.com/ghcjs/ghcjs-boot does not build with GHC 7.8.4, although the released version of transformers-0.4.3.0 does (commit ad784bdb801116d396d6f0f2020d4ece5b26f79e is the last to build, and a10a5b0bba4e720b1d29b621856ad851565c9146 breaks it).

**Comment by BethAr** [comment]
I managed to pass the single-file package database issue by updating Cabal manually from the 1.22 branch. However, I got stuck at a later point and opened a new issue.
